lms/models/_lms.py

```diff
@@ -0,0 +1,21 @@
+import sqlalchemy as sa
+
+from lms.db import BASE
+
+
+class LMS(BASE):
+    __tablename__ = "lms"
+    __table_args__ = (
+        sa.UniqueConstraint("application_instance_id", "tool_consumer_instance_guid"),
+    )
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    application_instance_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("application_instances.id", ondelete="cascade"),
+    )
+
+    tool_consumer_instance_guid = sa.Column(sa.String, nullable=False)
+
+    tool_consumer_info_product_family_code = sa.Column(sa.UnicodeText(), nullable=True)
```

Comment:
**Update:** I think we want a nullable, one-to-one relationship between `ApplicationInstance` and `LMS`. Not a one-to-many as you have it here.

1. When an `ApplicationInstance` is first created (when the user submits the https://lms.hypothes.is/welcome form) the `ApplicationInstance` hasn't been installed in an LMS or launched by an LMS yet. At this time `ApplicationInstance.lms_id` is `NULL`
2. The first time we receive an LTI launch for an `ApplicationInstance` whose `lms_id` is `NULL` we create a new `LMS` based on the launch params and set `ApplicationInstance.lms_id` to that LMS's ID. That `ApplicationInstance` is now "locked" to that LMS. Note that the `LMS` table is grouped by `ApplicationInstance`: if multiple `ApplicationInstance`'s are installed in the same LMS that LMS will have multiple entries in the `LMS` table
3. In the future (separate PR) we'll make it so that if we receive an LTI launch for an `ApplicationInstance` and the `tool_consumer_instance_guid` in the launch params doesn't match the `ApplicationInstance`'s related `LMS.tool_consumer_instance_guid` we refuse to launch and show an error message. You cannot install the same `ApplicationInstance` in multiple LMS's or change your LMS's GUID. We're going to show an error message in that case